reference/docs-conceptual/learn/ps101/06-flow-control.md

```diff
@@ -0,0 +1,336 @@
+Too low!
+What's your guess?: 2
+Too low!
+What's your guess?: 3
+```
+
+The previous example is a numbers game that continues until the value you guess equals the same
+number that the `Get-Random` cmdlet generated.
+
+`Do While` is just the opposite. It runs as long as the specified condition evaluates to true.
+
+```powershell
+$number = Get-Random -Minimum 1 -Maximum 10
+do {
+>>     $guess = Read-Host -Prompt "What's your guess?"
+>>     if ($guess -lt $number) {
+>>         Write-Output 'Too low!'
+>>     }
+>>     elseif ($guess -gt $number) {
+>>         Write-Output 'Too high!'
+>>     }
+>> }
+>> while ($guess -ne $number)
+```
+
+```Output
+What's your guess?: 1
+Too low!
+What's your guess?: 2
+Too low!
+What's your guess?: 3
+Too low!
+What's your guess?: 4
+```
+
+The same results are achieved with a `Do While` loop by reversing the test condition to not equals.
+
+`Do` loops always run at least once because the condition is evaluated at the end of the loop.
+
+### While
+
+Similar to the `Do While` loop, a `While` loop runs as long as the specified condition is true. The
+difference however, is that a `While` loop evaluates the condition at the top of the loop before any
+code is run. So it doesn't run if the condition evaluates to false.
+
+```powershell
+$date = Get-Date -Date 'November 22'
+while ($date.DayOfWeek -ne 'Thursday') {
+>>     $date = $date.AddDays(1)
+>> }
+Write-Output $date
+```
+
+```Output
+Thursday, November 23, 2017 12:00:00 AM
+```
+
+The previous example calculates what day Thanksgiving Day is on in the United States. It's always on
+the fourth Thursday of November. So the loop starts with the 22nd day of November and adds a day
+while the day of the week isn't equal to Thursday. If the 22nd is a Thursday, the loop doesn't run
+at all.
+
+## Break, Continue, and Return
+
+`Break` is designed to break out of a loop. It's also commonly used with the `switch` statement.
+
+```powershell
+for ($i = 1; $i -lt 5; $i++) {
+>>     Write-Output "Sleeping for $i seconds"
+>>     Start-Sleep -Seconds $i
+>>     break
+>> }
+```
+
+```Output
+Sleeping for 1 seconds
+```
+
+The `break` statement shown in the previous example causes the loop to exit on the first iteration.
+
+Continue is designed to skip to the next iteration of a loop.
+
+```powershell
+while ($i -lt 5) {
+>>     $i += 1
+>>     if ($i -eq 3) {
+>>         continue
+>>     }
+>>     Write-Output $i
+>> }
+```
+
+```Output
+1
+2
+4
+5
+```
+
+The previous example will output the numbers 1, 2, 4, and 5. It skips number 3 and continues with
+the next iteration of the loop. Similar to `break`, `continue` breaks out of the loop except only
+for the current iteration. Execution continues with the next iteration instead of breaking out of
+the loop and stopping.
+
+Return is designed to exit out of the existing scope.
+
+```powershell
+$number = 1..10
+foreach ($n in $number) {
+>>     if ($n -ge 4) {
+>>         Return $n
+>>     }
+>> }
+```
```

Comment:
```suggestion
```powershell
$number = 1..10
foreach ($n in $number) {
  if ($n -ge 4) {
    Return $n
  }
}
```
```

Reply:
Fixed